skmultiflow/classification/trees/hoeffding_adaptive_tree.py

```diff
@@ -0,0 +1,489 @@
+        X: Numpy.ndarray of shape (n_samples, n_features)
+            Data instances.
+
+        y: Array-like
+            Contains the classification targets for all samples in X.
+
+        classes: Not used.
+
+        weight: Float or Array-like
+            Instance weight. If not provided, uniform weights are assumed.
+
+        Returns
+        -------
+        self
+
+        """
+        if y is not None:
+            if weight is None:
+                weight = np.array([1.0])
+            row_cnt, _ = get_dimensions(X)
+            wrow_cnt, _ = get_dimensions(weight)
+            if row_cnt != wrow_cnt:
+                weight = [weight[0]] * row_cnt
+            for i in range(row_cnt):
+                if weight[i] != 0.0:
+                    self._partial_fit(X[i], y[i], weight[i])
+
+    # Override HoeffdingTree
+    def _partial_fit(self, X, y, weight):
+
+        if self._tree_root is None:
+            self._tree_root = self._new_learning_node()
+            self._active_leaf_node_cnt = 1
+        self._tree_root.learn_from_instance(X, y, weight, self, None, -1)
+
+    def filter_instance_to_leaves(self, X, split_parent, parent_branch, update_splitter_counts):
+        nodes = []
+        self._tree_root.filter_instance_to_leaves(X, split_parent, parent_branch, update_splitter_counts, nodes)
+        return nodes
+
+    # Override HoeffdingTree
+    def get_votes_for_instance(self, X):
+        if self._tree_root is not None:
+            found_node = self.filter_instance_to_leaves(X, None, -1, False)
+            result = {}
+            predict_path = 0
+            for fn in found_node:
+                if fn.parent_branch != -999:
+                    leaf_node = fn.node
+                    if leaf_node is None:
+                        leaf_node = fn.parent
+                    dist = leaf_node.get_class_votes(X, self)
+                    result.update(dist)  # add elements to dictionary
+            return result
+        else:
+            return {}
+
+    # Override HoeffdingTree/BaseClassifier
+    def predict(self, X):
+        r, _ = get_dimensions(X)
+        predictions = []
+        for i in range(r):
+            votes = self.get_votes_for_instance(X[i])
+            if votes == {}:
+                # Tree is empty, all classes equal, default to zero
+                predictions.append(0)
+            else:
+                predictions.append(max(votes, key=votes.get))
+        return predictions
+
+    def score(self, X, y):
+        raise NotImplementedError
+
+    def predict_proba(self, X):
+        raise NotImplementedError
+
+    # Override HoeffdingTree
+    def _new_learning_node(self, initial_class_observations=None):
+        return self.AdaLearningNode(initial_class_observations)
+
+    # Override HoeffdingTree
+    def new_split_node(self, split_test, class_observations, size):
+        return self.AdaSplitNode(split_test, class_observations, size)
+
+# Example TODO Create a demo/test from this
```

Comment:
- [ ] Create a demo/test from this.
- [ ] Example can be kept for reference. In that case, add it to the docstring of HAT under the subsection **Example**

Reply:
Done!

Reply:
OK. The demo will be included in next batch of changes.